makefile

```diff
@@ -7,23 +7,16 @@ ifeq ($(XINERAMA),true)
 	XINERAMA_FLAGS = --all-features # idk if there will ever be a workaround
 endif
 
-ifeq ($(CC),)
-	CC = cc
-endif
-
```

Comment:
I didn't realize what failure you were talking about until returning to this just today, because `CC` happened to be set in my environment without my realizing and everything I ran seemed to work.

I've restored the bits of the `make` build that set `CC` appropriately so that the `config` subproject and `checkdeps.sh` script will work.

I agree it should be removed and would have liked to follow the nice plan you laid out, I just don't have the time or energy at the moment and wanted to get this PR into a merge-able state.

Will keep it in mind though if I find some time to work on this project again! That is, if you don't get to it first. :)